django_fido/management/commands/download_authenticator_metadata.py

```diff
@@ -114,37 +114,64 @@ class Command(BaseCommand):
     def add_arguments(self, parser):
         """Parse command arguments."""
 
-    def handle(self, **options):
-        """Donwload and parse metadata from metadata service."""
-        try:
-            SETTINGS.metadata_service
-        except TypeError:
-            raise CommandError('access_token setting must be specified for this command to work.')
+    def _update_auth(self, authenticator_data: Dict[str, Any], identifier: List[str],
+                     mds3: bool, hash_alg: str) -> List[str]:
+        """Update individual authenticators from MDS."""
+        downloaded_identifiers = []
+        if not mds3:
+            url = authenticator_data['url']
+            # Cleanup the old instances
+            try:
+                authenticator = AuthenticatorMetadata.objects.get(url=url)
+            except AuthenticatorMetadata.DoesNotExist:
+                pass
+            else:
+                authenticator.delete()
+            auth_metadata = requests.get(url, params={'token': SETTINGS.metadata_service['access_token']},
+                                         timeout=SETTINGS.metadata_service['timeout'])
+            hash = hashlib.new(hash_alg, auth_metadata.content)
+            if hash.digest() != urlsafe_b64decode(authenticator_data['hash'].encode()):
+                self.stderr.write('Hash invalid for authenticator {}.'.format(identifier))
+                detailed_metadata_entry = ''
+            else:
+                detailed_metadata_entry = urlsafe_b64decode(auth_metadata.content).decode()
+        for ident in identifier:
+            downloaded_identifiers += [ident]
+            authenticator, _ = AuthenticatorMetadata.objects.get_or_create(identifier=ident)
+            authenticator.metadata_entry = json.dumps(authenticator_data)
+            if not mds3:
+                authenticator.detailed_metadata_entry = detailed_metadata_entry
+            authenticator.save()
+        return downloaded_identifiers
 
+    def handle(self, **options):
+        """Download and parse metadata from metadata service."""
```

Comment:
Since now it not just adds new content, but also deletes old data, transaction management should be added to prevent unwanted state upon encountering an error.